### ISSUE TITLE
Reduce data holdings in this repository

### ISSUE THREAD

**Issue by matt-long** [CESR-lab/ucla-roms#26]
The size of the ROMS repository is somewhat unwieldy, with large data collections, for example, in [Examples/input_data](https://github.com/CESR-lab/ucla-roms/tree/main/Examples/input_data). We're interested in rapidly deploying ROMS and this large size reduces flexibility for cloning patterns. 

These datasets are likely to change with a lower temporal frequency than the source code, so we're carrying around a lot of extra overhead. 

I propose that we consider a redesign of the data holding, likely pulling them out into a separate `ROMS-Inputdata` repository, for example, or exploring other solutions (e.g., https://dvc.org/).

**Comment by dafyddstephenson** [comment]
Good idea, Matt. The `input_data` directory isn't too bad, but I believe the files in it have been changed several times, and git cannot keep track of diffs in netcdf files, so instead keeps multiple versions of very similar files in its history, which are *all* copied to the local system on `git clone`. If uploading the files to a fresh repo with a clean history, it would carry a much lower footprint, as long as the files aren't changed going forward. However, the files would need to be deep cleaned out of the current repo's history.

**Comment by dafyddstephenson** [comment]
@nmolem following our chat earlier, I used the BFG repo cleaner
Here are the steps I followed:

1. Clone a clean copy of the existing repo somewhere as a backup
e.g. `git clone https://github.com/CESR-lab/ucla-roms.git roms_tmp/` 

If I run `du -sch *` on this repo I get :
```
 13M	Documentation
3.3M	Examples
 36K	LICENSE
3.4M	Legacy
1.9M	NHMG
1.3M	Tools-Roms
4.0K	Work
4.0K	readme
2.0M	src
 25M	total
```
However, `du -sch .*` shows:
```
417M	.git
4.0K	.gitignore
417M	total
```
revealing that ~95% of the data are in the repo history.

2. Clone a "bare" version of the repo using `--mirror` (on which we'll do the cleaning):
e.g. `git clone --mirror https://github.com/CESR-lab/ucla-roms.git roms_clean.git`

Before:
```
du -sh roms_clean.git 
417M	roms_clean.git
```
3. Get the BFG repo cleaner from : https://rtyley.github.io/bfg-repo-cleaner/ (there's a download link on the right that downloads a java file)

4. run BFG to strip out all netcdf files (you can also use e.g. `--delete-folders` to remove any particular folder instead)
e.g. `java -jar ~/Downloads/bfg-1.14.0.jar --delete-files "*.nc" roms_clean.git`

5. Clean up old reference logs, e.g.:
```
$ cd roms_clean.git
$ git reflog expire --expire=now --all && git gc --prune=now --aggressive
```
After:
```
du -sh roms_clean.git
 20M	roms_clean.git
```

6. Push up the changes

After doing this on a fork of ROMS (which I've pushed on my other account to https://github.com/ds4g15/ucla-roms as an example), then cloning it again (without `--mirror`) I saw the size drop to 48MB:
```
du -sch *
 13M	Documentation
3.3M	Examples
 36K	LICENSE
3.4M	Legacy
1.9M	NHMG
1.3M	Tools-Roms
4.0K	Work
4.0K	readme
2.0M	src
 25M	total
```
as before, but now:
```
du -sch .*
 23M	.git
4.0K	.gitignore
 23M	total
```

...If anything goes wrong you can fall back on the backup repo

**Comment by nmolem** [comment]
I believe that I did it successfully. I used the filter-repo command which was pretty painless.

**Comment by dafyddstephenson** [comment]
seems to have worked! I'll try running the Examples etc. with [the netcdf data in the Box](https://ucla.box.com/s/tu6must5svcz8e5ebvgwc7t5dgjeq9bt) and keep an eye out for any commits from you in the next day or so. Cheers
